### PR TITLE
Improve enabling openacc test for supported compilers

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

File: tests/regression/array/CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,13 @@ target_link_libraries(gen_regression_array_driver_fortran gen_regression_array_f
 add_test(NAME gen_regression_array_driver_fortran
     COMMAND $<TARGET_FILE:gen_regression_array_driver_fortran>)
 
-if(CUDA_AVAILABLE AND "${CMAKE_CXX_COMPILER_VERSION}" VERSION_GREATER_EQUAL "6")
```
Comment (havogt) [inline tests/regression/array/CMakeLists.txt:8]: What was your idea with the original condition? I guess selecting gnu >=6?

Comment (lukasm91) [reply tests/regression/array/CMakeLists.txt:8]: Yes :)

File: tests/regression/array/CMakeLists.txt [hunk 2]
```diff
@@ -5,7 +5,13 @@ target_link_libraries(gen_regression_array_driver_fortran gen_regression_array_f
...
+if( CUDA_AVAILABLE
+    AND ${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" # -fopenacc will be passed to the c++ linker
```
Comment (lukasm91) [inline tests/regression/array/CMakeLists.txt]:
```suggestion
    AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" # -fopenacc will be passed to the c++ linker
```

File: tests/regression/array/CMakeLists.txt [hunk 3]
```diff
@@ -5,7 +5,13 @@ target_link_libraries(gen_regression_array_driver_fortran gen_regression_array_f
...
+    AND "${CMAKE_CXX_COMPILER_VERSION}" VERSION_GREATER_EQUAL "6"
+    AND ${CMAKE_Fortran_COMPILER_ID} STREQUAL "GNU"
```
Comment (lukasm91) [inline tests/regression/array/CMakeLists.txt]:
```suggestion
    AND "${CMAKE_Fortran_COMPILER_ID}" STREQUAL "GNU"
```

File: tests/regression/array/CMakeLists.txt [hunk 4]
```diff
@@ -5,7 +5,13 @@ target_link_libraries(gen_regression_array_driver_fortran gen_regression_array_f
 add_test(NAME gen_regression_array_driver_fortran
     COMMAND $<TARGET_FILE:gen_regression_array_driver_fortran>)
 
-if(CUDA_AVAILABLE AND "${CMAKE_CXX_COMPILER_VERSION}" VERSION_GREATER_EQUAL "6")
+if( CUDA_AVAILABLE
```
Comment (lukasm91) [inline tests/regression/array/CMakeLists.txt:8]: formatting (I think we converged our code style to have no space after `if(`)

File: tests/regression/array/CMakeLists.txt (hunk 2 above)
Comment (lukasm91) [reply tests/regression/array/CMakeLists.txt]: we could also use `GNU` instead of `"GNU"` but I have no preference there

**Review by lukasm91 — CHANGES_REQUESTED** [review]

File: tests/regression/array/CMakeLists.txt (hunk 4 above)
Comment (havogt) [reply tests/regression/array/CMakeLists.txt:8]: Yes, here I chose it to align all conditions, i.e. it is not a space but a tab which is a space.

**Comment by havogt** [comment]
launch jenkins

**Review by lukasm91 — APPROVED** [review]

**Pull request merged**